### PR TITLE
fix(web_api): add type stubs for stripe

### PR DESCRIPTION
**Pull request by sbdchd**
Latest version of stripe type stubs are more filled out and provide some useful type checking:
https://github.com/python/typeshed/pull/7978

### CODE REVIEW

File: web_api/pyproject.toml [hunk 1]
```diff
@@ -41,6 +41,9 @@ types-redis = "^0.1.10"
 types-dj-database-url = "^0.5.0"
 responses = "^0.21.0"
 pytest-django = "^4.5.2"
+types-stripe = "^2.59.10"
+pydantic = "^1.9.1"
+types-requests = "^2.27.30"
```
Comment (sbdchd) [inline web_api/pyproject.toml:46]: mypy no longer ships the typeshed so we need to specify what we want

File: web_api/web_api/settings.py [hunk 2]
```diff
@@ -15,7 +15,8 @@
 patch_django()
 
 
-sentry_sdk.init(integrations=[DjangoIntegration()], send_default_pii=True)
+# Waiting on https://github.com/getsentry/sentry-python/issues/1415
+sentry_sdk.init(integrations=[DjangoIntegration()], send_default_pii=True)  # type: ignore[abstract]
```
Comment (sbdchd) [inline web_api/web_api/settings.py:19]: waiting on https://github.com/getsentry/sentry-python/issues/1415

File: web_api/pyproject.toml [hunk 3]
```diff
@@ -41,6 +41,9 @@ types-redis = "^0.1.10"
 types-dj-database-url = "^0.5.0"
 responses = "^0.21.0"
 pytest-django = "^4.5.2"
+types-stripe = "^2.59.10"
+pydantic = "^1.9.1"
```
Comment (sbdchd) [inline web_api/pyproject.toml:45]: updating due to mypy issue: https://github.com/samuelcolvin/pydantic/issues/3528

File: web_api/pyproject.toml [hunk 4]
```diff
@@ -24,7 +24,7 @@ pywatchman = "^1.4"
 [tool.poetry.dev-dependencies]
 pylint = "^2.4"
 pytest = "6.0.1"
-mypy = "^0.812"
+mypy = "^0.960"
```
Comment (sbdchd) [inline web_api/pyproject.toml:27]: updated so mypy understandings `type[T]`

**Review by chdsbd — APPROVED** [review]

**Pull request merged**